### PR TITLE
[#87] Add documentation for Future and IO monads

### PR DESCRIPTION
**Pull request by christobill**
Relates to https://github.com/scalikejdbc/scalikejdbc.github.io/issues/87

### CODE REVIEW

File: source/documentation/future.html.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+---
```
Comment (seratch) [inline source/documentation/future.html.md]: We don't need this page as it's already available on https://github.com/scalikejdbc/scalikejdbc.github.io/blob/develop/source/documentation/transaction.html.md

File: source/documentation/io-monads.html.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
+---
+title: Working with IO monads - ScalikeJDBC
```
Comment (seratch) [inline source/documentation/io-monads.html.md]: Can you add this part to https://github.com/scalikejdbc/scalikejdbc.github.io/blob/develop/source/documentation/transaction.html.md

File: source/layouts/layout.erb [hunk 3]
```diff
@@ -54,6 +54,8 @@
           <p><%= link_to 'Logging / Query Inspector', '/documentation/query-inspector.html' %></p>
           <p><%= link_to 'Reverse Engineering', '/documentation/reverse-engineering.html' %></p>
           <p><%= link_to 'Testing', '/documentation/testing.html' %></p>
+          <p><%= link_to 'Working with Future', '/documentation/future.html' %></p>
```
Comment (seratch) [inline source/layouts/layout.erb]: Please remove these changes

File: source/documentation/future.html.md (hunk 1 above)
Comment (christobill) [reply source/documentation/future.html.md]: Yes true! I removed this page

File: source/documentation/io-monads.html.md (hunk 2 above)
Comment (christobill) [reply source/documentation/io-monads.html.md]: Done

File: source/layouts/layout.erb (hunk 3 above)
Comment (christobill) [reply source/layouts/layout.erb]: Done

**Comment by christobill** [comment]
@seratch thanks a lot!

**Pull request merged**